net/can/can_recvmsg.c

```diff
@@ -344,7 +344,10 @@ static uint16_t can_recvfrom_eventhandler(FAR struct net_driver_s *dev,
            * when is valid then complete the read action.
            */
 #ifdef CONFIG_NET_CANPROTO_OPTIONS
-          if (can_recv_filter(conn, (canid_t) *dev->d_appdata) == 0)
+          canid_t can_id;
```

Comment:
Maybe we can reconstruct can-id by shit and or operations instead of memcpy?

Reply:
> Maybe we can reconstruct can-id by shit and or operations instead of memcpy?

like this : 
canid_t can_id = dev->d_appdata[0] << 24  |  
                           dev->d_appdata[1] << 16  | 
                           dev->d_appdata[2] << 8    | 
                           dev->d_appdata[2] 
?

Reply:
Yes. Let me search, there should be already a macro for it

Reply:
:( No, unfortunately there is only one for IP `ip6_get_ipv4addr`. Let's move with current change unless you want to introduce similar macro for `canid_t`